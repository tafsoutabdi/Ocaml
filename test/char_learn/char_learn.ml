open Tokenization.Characters.Tokenizer
open Utils.Characters
open Utils.Std

let runtest batch = 
  Format.printf "batch = %s@.learn batch = %s@.@."
    (string_of_list string_of_string batch)  
    (string_of_vocabulary (learn batch))

let () = runtest ["facile"]

let () = runtest ["abracadabra"; "pffouit"]

let () = runtest ["difficile"; "pas facile"]
    