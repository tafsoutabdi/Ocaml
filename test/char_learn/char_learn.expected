batch = ["facile"]
learn batch = ['f'; 'a'; 'c'; 'i'; 'l'; 'e']

batch = ["abracadabra"; "pffouit"]
learn batch = ['a'; 'b'; 'r'; 'c'; 'd'; 'p'; 'f'; 'o'; 'u'; 'i'; 't']

batch = ["difficile"; "pas facile"]
learn batch = ['d'; 'i'; 'f'; 'c'; 'l'; 'e'; 'p'; 'a'; 's'; ' ']

